Removed and replaced the XML window with an ImGui one

grimmier378 opened the pull request:
* Removed the XML in game window.
* Added ImGui Window (same as the settings panel)
* Reworked the /pluginman command to toggle ImGui Window open | closed.
* Tossed a toggle button on the settings panel to toggle the main window.

grimmier378 committed:
Removed and replaced the XML window with an ImGui one

* Removed the XML in game window.
* Added ImGui Window (same as the settings panel)
* Reworked the /pluginman command to toggle ImGui Window open | closed.
* Tossed a toggle button on the settings panel to toggle the main window.

grimmier378 committed:
removed the toggle button

removed the toggle button and HelpMarker and replaced with simple Text explaining /pluginman command

brainiac commented on MQ2PluginManager.cpp:
```diff
@@ -4,30 +4,37 @@
 	around until things worked. There may be some residual code that is not even required by the
 	plugin in here, but if it is, I'm unaware of it. Chances are some of these things I just deleted
 	entirely and tried to run after a succesful compile only to crash and therefore was readded.
+
+	July  9, 2024 Grimmier  -- Added ImGui Window to the mqsettings Panel.
+	July 30, 2024 Grimmier  -- Removed the in game XML window and replaced with a IMGUI window.
+							-- Added a toggle button to the mqsettings panel window.
+							-- Switched the /pluginman command to toggle the IMGUI window instead of the old XML.
```
This is unnecessary

brainiac commented on MQ2PluginManager.cpp:
```diff
@@ -4,30 +4,37 @@
 #include "CPluginTree.h"
-#include "CPluginToolWnd.h"
+#include "imgui/ImGuiUtils.h"
+#include <imgui_internal.h>
```
Why is the internal header being included here?

grimmier378 replied:
oops i was using easyfind plugin as a check to make sure i was creating the standalone window correctly.

brainiac commented on MQ2PluginManager.cpp:
```diff
@@ -161,20 +129,29 @@ void DrawPluginManager_MQSettingsPanel()
+PLUGIN_API void OnUpdateImGui()
 {
-	CreatePluginWindow();
+	if (GetGameState() == GAMESTATE_INGAME)
```
I see no reason why this needs to be tied to game state anymore.

brainiac reviewed (commented):
I left some comments/questions

grimmier378 committed:
addresses comments

* Added MQ2PluginManager.ini setting to save window state.
* Window will show at load the first time.
* Setting saves when display state changes.
* Removed game state check

Sicprofundus commented on MQ2PluginManager.cpp:
```diff
@@ -4,30 +4,37 @@
 PLUGIN_API void InitializePlugin()
 {
-	WriteChatColor("\aoLoading MQ2PluginManager v0.0.2 ...");
+	WriteChatColor("\aoLoading MQ2PluginManager v0.0.3 ...");
```
if we're going to leave the version on it here, we should put the version somewhere and then use it here

grimmier378 replied:
dropped it since it really doesn't do anything

grimmier378 committed:
dropped the version number from initialize output.

the version number really did nothing as it was just a string output on initialize.

grimmier378 committed:
save state when manually closing window

this way the state is saved both manually closing or using the toggle command.

the prior was only saving the state with the slash command.

brainiac commented on MQ2PluginManager.cpp:
```diff
@@ -8,28 +8,30 @@
 #include <mq/Plugin.h>
 #include "CPluginTree.h"
-#include "CPluginToolWnd.h"
+#include "imgui/ImGuiUtils.h"
```
What is using this header?

grimmier378 replied:
doh nothing it appears.

brainiac commented on MQ2PluginManager.cpp:
```diff
@@ -161,20 +124,27 @@ void DrawPluginManager_MQSettingsPanel()
+	if (!s_showWindow)
+		return;
 
-PLUGIN_API void SetGameState(int GameState)
```
Changed my mind about this a little bit. Perhaps looking for additional feedback on this...

I don't know if we want to be immediately showing this at login by default, but would want it to appear in game by default like it did before. So, maybe make it toggleable at any point in the game, but only appear based base on last state when it was in game?

grimmier378 commented:
Yeah I was kind of thinking the same as I saw it appearing at login.

I can toss that check back in.

grimmier378 committed:
readded the game state check

brainiac commented:
> Yeah I was kind of thinking the same as I saw it appearing at login. I can toss that check back in.

Well I don't think you wrote all that I wrote. It would be nice if you could toggle the window from anywhere. The old gamestate check was because the in-game UI only works ingame. but ImGui works anywhere, so I don't think it should be as limited. Some thought needs to go into it.

grimmier378 committed:
removed unneeded header

grimmier378 commented:
how do i get the slash command to work at char select?

```> /plugin pluginmanager
Loading MQ2PluginManager...
/pluginman Toggles the PluginManager UI Window
Plugin 'MQ2PluginManager' loaded.
> /pluginman
DoCommand - Couldn't parse '/pluginman'
```

I have some idea's but can't get the toggle to work outside of the game.

brainiac commented:
> how do i get the slash command to work at char select?
> 
> ```
> Loading MQ2PluginManager...
> /pluginman Toggles the PluginManager UI Window
> Plugin 'MQ2PluginManager' loaded.
> > /pluginman
> DoCommand - Couldn't parse '/pluginman'
> ```
> 
> I have some idea's but can't get the toggle to work outside of the game.

Remove the optional params from AddCommand:
```
AddCommand("/Pluginman", PluginManagerCommand, false, false, true);
```

from the doc comments:
```c++
/**
 * Adds a new chat command. If the command conflicts with an existing EverQuest command, then
 * this command will take precedence. If the ocmmand conflicts with another MacroQuest command,
 * then this call will fail.
 *
 * @param command The name of the command to register, including a leading forward slash, e.g. "/hello"
 * @param handler The function that will be call when the command is executed.
 * @param eq If true, the handler is ignored and the command will be forwarded to EQ.
 * @param parse If true, any macro expressions will be parsed before passing the text to the handler.
 * @param inGame If true, the command will only be registered while in the game world.
 * @return True if the command is successfully added.
 */
bool AddCommand(std::string_view command, MQCommandHandler handler, bool eq = false, bool parse = true, bool inGame = false);
```

grimmier378 committed:
addresses toggeling out of game

Does not automatically show out of game you have to toggle it.

opening and closing the window both in and out of game will save that state. 

Will automatically appear in game based on saved state.
